### PR TITLE
Added accelerated openssl installation support 

### PR DESCRIPTION
**Pull request by rickit2022**
- Added QAT installation support for accelerated OpenSSL using features as specified in #5248

### CODE REVIEW

File: perfkitbenchmarker/linux_packages/openssl.py [hunk 1]
```diff
@@ -24,3 +24,32 @@ def YumInstall(vm):
...
+  """Installs QAT engine for accelerated OpenSSL."""
+
+  vm.InstallPackages('autoconf build-essential libtool cmake cpuid libssl-dev pkg-config nasm')
+  
+  # Install Intel® Integrated Performance Primitives Cryptography lib  
+  vm.RemoteCommand('git clone https://github.com/intel/ipp-crypto.git')
+  vm.RemoteCommand('cd ipp-crypto')
+  vm.RemoteCommand('git checkout ippcp_2021.7.1')
+  vm.RemoteCommand('cd ipp-crypto/sources/ippcp/crypto_mb')
+  vm.RemoteCommand('cmake . -Bbuild -DCMAKE_INSTALL_PREFIX=/usr')
```
Comment (jellyfishcake) [inline perfkitbenchmarker/linux_packages/openssl.py:39]: Every remote command starts a new shell session. Therefore, you are back in home directory here and cannot directly call cmake.
To fix, merge line 38 and 39 by doing 
cd ipp-crypto/sources/ippcp/crypto_mb && cmake . -Bbuild -DCMAKE_INSTALL_PREFIX=/usr

File: perfkitbenchmarker/linux_packages/openssl.py [hunk 2]
```diff
@@ -24,3 +24,32 @@ def YumInstall(vm):
...
+
+def AptInstallQAT(vm):
+  """Installs QAT engine for accelerated OpenSSL."""
+
+  vm.InstallPackages('autoconf build-essential libtool cmake cpuid libssl-dev pkg-config nasm')
+  
+  # Install Intel® Integrated Performance Primitives Cryptography lib  
+  vm.RemoteCommand('git clone https://github.com/intel/ipp-crypto.git')
+  vm.RemoteCommand('cd ipp-crypto')
+  vm.RemoteCommand('git checkout ippcp_2021.7.1')
```
Comment (jellyfishcake) [inline perfkitbenchmarker/linux_packages/openssl.py:37]: Every remote command starts a new shell session. Therefore, you are back in home directory here and cannot directly call cmake.
To fix, merge line 36 and 37 by doing
cd cd ipp-crypto && git checkout ippcp_2021.7.1

File: perfkitbenchmarker/linux_packages/openssl.py [hunk 3]
```diff
@@ -24,3 +24,32 @@ def YumInstall(vm):
...
+  vm.RemoteCommand('cd ipp-crypto/sources/ippcp/crypto_mb')
+  vm.RemoteCommand('cmake . -Bbuild -DCMAKE_INSTALL_PREFIX=/usr')
+
+  # Install Intel® Multi-Buffer Crypto for IPsec Library lib
+  vm.RemoteCommand('git clone https://github.com/intel/intel-ipsec-mb.git')
+  vm.RemoteCommand('cd intel-ipsec-mb')
+  vm.RemoteCommand('git checkout v1.3')
```
Comment (jellyfishcake) [inline perfkitbenchmarker/linux_packages/openssl.py:44]: same issue here and in the next two commands.

File: perfkitbenchmarker/linux_packages/openssl.py [hunk 4]
```diff
@@ -24,3 +24,32 @@ def YumInstall(vm):
...
+  vm.RemoteCommand('make -j')
+  vm.RemoteCommand('sudo make install NOLDCONFIG=y')
+
+  # Install QAT_Engine
+  vm.RemoteCommand('git clone https://github.com/intel/QAT_Engine.git')
+  vm.RemoteCommand('cd QAT_Engine')
+  vm.RemoteCommand('git checkout v1.2.0')
```
Comment (jellyfishcake) [inline perfkitbenchmarker/linux_packages/openssl.py:51]: same issue here and in the next 4 commands

**Review by jellyfishcake — CHANGES_REQUESTED** [review]

**Comment by rickit2022** [comment]
@jellyfishcake Hi thanks for the comments, I had no idea about that behavior and have now put all commands on the same call(s). Please let me know if there's anything else I need to change.

**Review by jellyfishcake — APPROVED** [review]

**Comment by jellyfishcake** [comment]
Looks good, thank you!

**Pull request merged**